syntax = "proto3";
package fedvision.paddle_fl;

message PaddleFLAggregatorTask {
  string scheduler_ep = 1;
  uint32 worker_num = 3;
  uint32 max_iter = 4;
  bytes main_program = 5;
  bytes startup_program = 6;
}


message PaddleFLWorkerTask {
  string scheduler_ep = 1;
  uint32 trainer_id = 2;
  string trainer_ep = 3;
  string entrypoint = 4;
  bytes main_program = 5;
  bytes startup_program = 6;
  bytes send_program = 7;
  bytes recv_program = 8;
  bytes feed_names = 9;
  bytes target_names = 10;
  bytes strategy = 11;
  string config_string = 12;
}

